unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/Editor/Procedural Skybox/SkyboxEditor.cs

```diff
@@ -0,0 +1,1344 @@
+
+            GUIStyle tStyle = new GUIStyle();
+            tStyle.alignment = TextAnchor.MiddleCenter;
+            tStyle.margin = new RectOffset(150, 200, 0, 0);
+            GUILayout.Label("Loaded: " + selectedConfiguration.skyboxID);
+            GUILayout.BeginHorizontal(tStyle);
+
+            if (creatingNewConfig)
+            {
+                GUILayout.Label("Name");
+                newConfigName = EditorGUILayout.TextField(newConfigName, GUILayout.Width(200));
+
+                if (GUILayout.Button("Create", GUILayout.Width(50)))
+                {
+                    // Make new configuration
+                    selectedConfiguration = AddNewConfiguration(newConfigName);
+
+                    // Update configuration list
+                    UpdateConfigurationsList();
+                    creatingNewConfig = false;
+                }
+
+                if (GUILayout.Button("Cancel", GUILayout.Width(50)))
+                {
+                    creatingNewConfig = false;
+                }
+            }
+            else
+            {
+                newConfigIndex = EditorGUILayout.Popup(selectedConfigurationIndex, configurationNames.ToArray(), GUILayout.Width(200));
+
+                if (newConfigIndex != selectedConfigurationIndex)
+                {
+                    selectedConfiguration = configurations[newConfigIndex];
+                    selectedConfigurationIndex = newConfigIndex;
+
+                    UpdateSlotsID();
+                }
+
+                if (GUILayout.Button("+", GUILayout.Width(50)))
+                {
+                    creatingNewConfig = true;
+                }
+            }
+
+
+            GUILayout.EndHorizontal();
+        }
+
+        private void UpdateConfigurationsList()
+        {
+            SkyboxConfiguration[] tConfigurations = Resources.LoadAll<SkyboxConfiguration>("Skybox Configurations/");
+            configurations = new List<SkyboxConfiguration>(tConfigurations);
+            configurationNames = new List<string>();
+
+            // If no configurations exist, make and select new one.
+            if (configurations == null || configurations.Count < 1)
+            {
+                selectedConfiguration = AddNewConfiguration("Generic Skybox");
+
+                configurations = new List<SkyboxConfiguration>();
+                configurations.Add(selectedConfiguration);
+            }
+
+            if (selectedConfiguration == null)
+            {
+                selectedConfiguration = configurations[0];
+            }
+
+            for (int i = 0; i < configurations.Count; i++)
+            {
+                configurationNames.Add(configurations[i].skyboxID);
+                if (selectedConfiguration == configurations[i])
+                {
+                    selectedConfigurationIndex = i;
+                }
+            }
+
+            UpdateSlotsID();
+
+            InitializeMaterial();
+
+            //isPaused = true;
```

Comment:
```suggestion
```

Remember that all commented code should be removed. For reverting code we already have the commits history!.

Reply:
Done. Making all these changes in a new commit